evap/evaluation/tests/test_models.py

```diff
@@ -719,6 +719,71 @@ def test_disable_cc(self):
         self.assertEqual(len(mail.outbox), 1)
         self.assertEqual(set(mail.outbox[0].cc), {self.additional_cc.email})
 
+    def test_send_contributor_publish_notifications(self):
+        responsible1 = baker.make(UserProfile)
+        responsible2 = baker.make(UserProfile)
+        # use is_single_result to get can_publish_average_grade to become true
+        evaluation1 = baker.make(Evaluation, course__responsibles=[responsible1], is_single_result=True)
+        evaluation2 = baker.make(Evaluation, course__responsibles=[responsible2])
+
+        editor1 = baker.make(UserProfile)
+        contributor1 = baker.make(UserProfile)
+
+        contributor2 = baker.make(UserProfile)
+        editor2 = baker.make(UserProfile)
+        contributor_both = baker.make(UserProfile)
+
+        # Contributions for evaluation1
+        baker.make(Contribution, evaluation=evaluation1, contributor=responsible1, role=Contribution.Role.CONTRIBUTOR)
+        baker.make(Contribution, evaluation=evaluation1, contributor=editor1, role=Contribution.Role.EDITOR)
+        baker.make(Contribution, evaluation=evaluation1, contributor=contributor1, role=Contribution.Role.CONTRIBUTOR)
+        baker.make(Contribution, evaluation=evaluation1, contributor=contributor_both, role=Contribution.Role.CONTRIBUTOR)
+
+        # Contributions for evaluation2
+        baker.make(
+            Contribution,
+            evaluation=evaluation2,
+            contributor=editor2,
+            role=Contribution.Role.EDITOR
+        )
+        contributor_both_contribution = baker.make(
+            Contribution,
+            evaluation=evaluation2,
+            contributor=contributor_both,
+            role=Contribution.Role.CONTRIBUTOR
+        )
+        contributor2_contribution = baker.make(
+            Contribution,
+            evaluation=evaluation2,
+            contributor=contributor2,
+            role=Contribution.Role.CONTRIBUTOR
+        )
```

Comment:
These lines are formatted differently than those for `evaluation1` and I can’t spot the difference :)

Reply:
![image](https://user-images.githubusercontent.com/13838962/90429119-5ce65700-e0c5-11ea-9a71-ce546fa9497f.png)
The main reason was that the upper 4 lines stay below 120 chars per line but the bottom three exceed that length significantly due to the variables that are assigned.

If i fully spread them all out, it will be 7 * 6 = 42 lines and I wanted to prevent that.

What do you think? Spread them all out?

Reply:
I solved it by adding two helpers, `make_contributor` and `make_editor` that I think are usable in a lot of other places too.